lib/internal/modules/esm/module_job.js

```diff
@@ -61,11 +60,14 @@ const CJSGlobalLike = [
   '__filename',
   '__dirname',
 ];
-const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
-  ArrayPrototypeSome(
-    CJSGlobalLike,
-    (globalLike) => errorMessage === `${globalLike} is not defined`,
-  );
+const getUndefinedCJSGlobalLike = (errorMessage) => {
+  for (const globalLike of CJSGlobalLike) {
+    if (errorMessage === `${globalLike} is not defined`) {
+      return globalLike;
+    }
+  }
+  return null;
+};
```

Comment:
I believe this can't use `for...of` (unsafe) and should remain using `ArrayPrototypeSome`